common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java

```diff
@@ -810,4 +814,66 @@ private void checkForDuplicateExternalId(List<Map<String, String>> list) {
       checkedList.add(externalId);
     }
   }
+
+  @SuppressWarnings("unchecked")
+  private void validateFrameworkDetails(Request request) {
+    if (request.getRequest().containsKey(JsonKey.FRAMEWORK)) {
+      if (!(request.getRequest().get(JsonKey.FRAMEWORK) instanceof Map)) {
+        throw new ProjectCommonException(
+            ResponseCode.dataTypeError.getErrorCode(),
+            ResponseCode.dataTypeError.getErrorMessage(),
+            ERROR_CODE,
+            JsonKey.FRAMEWORK,
+            JsonKey.MAP);
+      }
+      Map<String, Object> frameworkMap =
+          (Map<String, Object>) request.getRequest().get(JsonKey.FRAMEWORK);
+      validateParam(
```

Comment:
The user's framework ID needs to be inferred based on userId and association with root org or channel.